README.md

```diff
@@ -2,48 +2,149 @@
+* Exposed new developer filters.
+* Deduplicated code functionality.
+* Security: validate, sanitize, and escape (allthethings).
+
+### 3.0.1 ###
+* Fix browsing and uploading caused by broken nonce checks.
+* Normalize the SDK path for activation checks.
+
+### 3.0.0 ###
+* Security: fixes a bug that could allow unauthorized deletion of remotely-stored media.
+* Security: validate, sanitize, and escape (allthethings).
+* Security: use `https://` URLs by default, and warn if an insecure CNAME is configured.
+* Security: introduce permissions checks for specific actions within the plugin.
+* L10N/I18N: Round 1 of preparing strings for translation.
+* UI: Editor button more closely matches the WordPress Admin UI.
+* Add `srcset` to images added through the Media Library when Azure is the default media handler. (Props @patricknami).
+
+### 2.2 ###
+* Fixed network activation bug in WordPress multisite.
+* Fixed the issue with duplicate blob names in XML-RPC.
+
+### 2.1 ###
+* Fixed the issue with duplicate blob names.
+* Fixed the bug in uploading video files to blob storage.
+* Fixed the bug with forward slash in front of image thumbnail filenames.
+* Fixed the bug with year and month getting trimmed for file system images.
+
+### 2.0 ###
+* Updated to use Microsoft Azure SDK for PHP from https://github.com/WindowsAzure/azure-sdk-for-php and fixed to work with WordPress 3.4.1
+
+### 1.9 ###
+* Fixed case sensitivity error in file names on Linux
+
+### 1.8 ###
+* Bug fixed in generating blob storage URL when using Microsoft Azure Storage emulator
+
+### 1.7 ###
+* Added support to upload video files to blob storage
+
+### 1.5 ###
+* Included Microsoft Azure SDK for PHP v4.1.0 with the plugin. Now setting mime-type for uploaded file to blob storage.
+
+### 1.4 ###
+* Included Microsoft Azure SDK for PHP v4.0.2 with the plugin.
+
+### 1.3 ###
+* Included Microsoft Azure SDK for PHP v4.0.1 with the plugin, so no need to install the SDK separetely. Also fixed thumbnail handling issue while uploading files when some specific theme is enabled.
+
+### 1.2 ###
+* This release is compatible with Microsoft Azure SDK for PHP v3.0.0. It also fixes issue with deleting media files when thumbnails are associated.
+
+### 1.1 ###
+* This release is compatible with Microsoft Azure SDK for PHP v2.1.0 and WordPress 3.1
+
+### 1.0 ###
+* First release of Microsoft Azure Storage plugin for WordPress
+
+
+## Upgrade Notice ##
+
+### 3.0.0 ###
 This release features several security fixes and enhancements.
 It is highly recommended that all users upgrade immediately.
 
-## Support Level
-**Stable:** 10up is not planning to develop any new features for this, but will still respond to bug reports and security concerns.  We welcome PRs, but any that include new features should be small and easy to integrate and should not include breaking changes.  We otherwise intend to keep this tested up to the most recent version of WordPress.
```

Comment:
Will need to add this back so that the support level badge properly links to this section with clear details on what "stable" means